packages/pouch-storage/src/PouchStorageTable.ts

```diff
@@ -35,20 +35,18 @@ export type PouchDBSort = Array<
   string | { [propName: string]: 'asc' | 'desc' }
 >;
 
-// We may want to use `PouchDB.Find.ConditionOperators` instead of this, but
-// there are some mismatches in how we use this with the types.
-// https://github.com/deephaven/web-client-ui/issues/1505 to address this
 type PouchFilter = OnlyOneProp<{
   $eq: FilterValue | FilterValue[];
-  $neq: FilterValue | FilterValue[];
+  $ne: FilterValue | FilterValue[];
   $gt: FilterValue | FilterValue[];
   $gte: FilterValue | FilterValue[];
   $lt: FilterValue | FilterValue[];
   $lte: FilterValue | FilterValue[];
-  $regex: RegExp;
-}>;
+  $regex: RegExp | string;
+}> &
+  Omit<PouchDB.Find.ConditionOperators, '$regex'>;
```

Comment:
I was thinking about this after I suggested it, and I don't think the intersection type is actually providing anything useful. Namely, if you change `$ne` back to `$neq`, the compiler doesn't complain, so it's not really protecting anything. I would just remove the `& Omit<PouchDB.Find.ConditionOperators...` piece.